Source/WebCore/fileapi/Blob.cpp

```diff
@@ -288,19 +289,38 @@ void Blob::text(Ref<DeferredPromise>&& promise)
     });
 }
 
+static ExceptionOr<JSC::ArrayBuffer&> arrayBufferFromBlobLoader(BlobLoader& blobLoader)
+{
+    if (auto optionalErrorCode = blobLoader.errorCode())
+        return Exception { *optionalErrorCode };
+    RefPtr arrayBuffer = blobLoader.arrayBufferResult();
+    if (!arrayBuffer)
+        return Exception { ExceptionCode::InvalidStateError };
+    return *arrayBuffer;
+}
+
 void Blob::arrayBuffer(Ref<DeferredPromise>&& promise)
 {
     loadBlob(FileReaderLoader::ReadAsArrayBuffer, [promise = WTFMove(promise)](BlobLoader& blobLoader) mutable {
-        if (auto optionalErrorCode = blobLoader.errorCode()) {
-            promise->reject(Exception { *optionalErrorCode });
+        auto arrayBuffer = arrayBufferFromBlobLoader(blobLoader);
+        if (arrayBuffer.hasException()) {
+            promise->reject(arrayBuffer.releaseException());
             return;
         }
-        auto arrayBuffer = blobLoader.arrayBufferResult();
-        if (!arrayBuffer) {
-            promise->reject(Exception { ExceptionCode::InvalidStateError });
+        promise->resolve<IDLArrayBuffer>(arrayBuffer.releaseReturnValue());
```

Comment:
We can create a DOMPromiseDeferred<IDLArrayBuffer> from Ref<DeferredPromise>&& which could allow us to use `settle`. We may be able to do something like:
```
void Blob::arrayBuffer(DOMPromiseDeferred<IDLArrayBuffer>&& promise)
{
  ...
      promise.settle(arrayBufferFromBlobLoader(blobLoader));
  ...
}
```